index.sh

```diff
@@ -95,36 +159,134 @@ re-run this script.
+  elif [ $isatty -eq 1 ] ; then
+    cat 1>&2 << 'EOM'
+/=====================================\\
+|     update-alternatives detected     |
+\\=====================================/
+
+getmicro can use update-alternatives to register micro as a system text editor.
+For example, this will allow `crontab -e` open the cron file with micro.
+
+To avoid this prompt in the future, define the GETMICRO_REGISTER variable. E.x:
+
+  $ curl https://getmic.ro | GETMICRO_REGISTER=y sh
+  
+Many people find it useful to make micro available on the PATH. One way to do
+ this is to enter a root shell and run `cd /usr/bin` prior to installation.
+
+EOM
+    cpt="Register '$wrkdir/micro' with update-alternatives [y/n]: "
+    if command -v printf >/dev/null 2>&1 ; then
+      printf '%s' "$cpt" 1>&2
+    else
+      # Wrapping this in eval helps this script to pass shellcheck
+      eval '( echo -n "$cpt" 2>/dev/null || echo -e "$cpt"'\''\c'\'' 2>/dev/null || echo "$cpt" ) 1>&2'
+    fi
+    if command -v head >/dev/null 2>&1 ; then
+      # needed when piping curl into sh as its a subshell so one must reopen the tty
+      doRegister="$(head -n1 /dev/tty)"
+    elif command -v sed >/dev/null 2>&1 ; then
+      doRegister="$(sed 1q)"
+    else
+      read -r doRegister
+    fi
+    echo # add new line after long message and user input for prettier output
+  else
+    # default to not installing
+    doRegister="n"
+  fi
+  
+  if [ "${doRegister:-x}" = "y" ] ; then
+    if [ -w /etc/alternatives ] ; then # if we have write permission to /etc/alternatives
+      # hope we are effectively running as root
+      echo "Installing '$wrkdir/micro' as /usr/bin/editor..."
+      $altcmd --install /usr/bin/editor editor "$wrkdir/micro" 80
+
+      if command -v git >/dev/null 2>&1 ; then
+        # set the absolute lowest priority default value of core.editor to be /usr/bin/editor
+        # note that this git config will error out if the config key is already set
+        if git config --system --path core.editor /usr/bin/editor >/dev/null 2>&1 ; then
+          echo "Configuring git to use /usr/bin/editor as the default core editor..."
+        fi
+      fi
+
+      if [ -w /etc/environment ] && ! grep -qi '^EDITOR=' /etc/environment ; then
+        # set the absolute lowest priority default value of EDITOR to be /usr/bin/editor
+        echo "Configuring /etc/environment to use /usr/bin/editor as the default text EDITOR..."
+        echo 'EDITOR=/usr/bin/editor' >> /etc/environment 
+      fi
+      echo # pretty print new line to separate sections
+    else
+      cat 1>&2 << 'EOM'
+/=====================================\\
+|       INSUFFICIENT PRIVILEGES       |
+\\=====================================/
+
+Uh oh! We couldn't run update-alternatives due to insufficient privileges.
+
+To continue, try running getmicro as root or another privileged user. Examples:
+
+  $ curl https://getmic.ro | sudo sh
+  
+Or:
+
+  $ su - root -c "wget -O- https://getmic.ro | sh"
```

Comment:
I don't think this works -- the shell parses this as `(su - root -c "wget -O- https://getmic.ro) | (sh)` (so `sh` still gets run as the current user).

Reply:
I'm 100% certain that special characters are ignored inside a command:

```sh
echo "this && that | pipe > to < from || this"
```

will print:

```
this && that | pipe > to < from || this
```

Reply:
Ah my bad, I think i was just misreading where the quotes were.